questionnaire/task-list/task-list-service.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+function createTaskListService() {
+    // TODO: where should this live?
+    const taskStatusPrecedence = [
+        'notApplicable',
+        'cannotStartYet',
+        'completed',
+        'incomplete',
+        'applicable'
+    ];
+
+    // TODO: where should this live?
+    const taskAvailableStatus = ['completed', 'incomplete', 'applicable'];
+
+    function isTaskListSchema({sectionSchema, sectionId} = {}) {
+        if (sectionSchema) {
+            const pageType = sectionSchema?.meta?.pageType;
+            if (pageType === 'task-list') {
+                return true;
+            }
+        }
+
+        // TODO: pass in the sectionSchema, not the sectionId. OR
+        // TODO: get schema from sectionId and interrogate the shape instead of the IDs value.
+        if (sectionId) {
+            if (sectionId === 'p-task-list') {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    function getInitialPageIdOfTask(taskId, questionnaireDefinition) {
+        return questionnaireDefinition.routes.states[taskId].initial;
+    }
+
+    function decoratePageId(pageId) {
+        if (pageId.startsWith('p--')) {
+            return pageId.replace(/^p--/, 'info-');
+        }
+        return pageId.replace(/^p-/, '');
+    }
+
+    function getTaskStatus(taskId, questionnaireDefinition) {
+        const taskCompletionStatus =
+            questionnaireDefinition.routes.states?.[`${taskId}__completion-status`]
+                .currentSectionId;
+        const taskApplicabilityStatus =
+            questionnaireDefinition.routes.states?.[`${taskId}__applicability-status`]
+                .currentSectionId;
+
+        // the lower the index, the higher the precedent.
+        if (
+            taskStatusPrecedence.indexOf(taskCompletionStatus) <
+            taskStatusPrecedence.indexOf(taskApplicabilityStatus)
+        ) {
+            return taskCompletionStatus;
+        }
+
+        return taskApplicabilityStatus;
+    }
```

Comment:
I understand what you've done here, but I think it should be:

```
const status = taskApplicabilityStatus = 'applicable' ? taskCompletionStatus : taskApplicabilityStatus
```
It's not a perfect solution but I think it's just far easier to read and understand than these precedence arrays.